azurelinuxagent/ga/monitor.py

```diff
@@ -193,3 +193,13 @@ def daemon(self):
             except Exception as e:
                 logger.warn("Failed to send events: {0}", e)
             time.sleep(60)
+
+    def add_sysinfo(self, event):
+        event_param_dict = {}  # key: TelemetryEventParam.name, val: idx of the EventParam in event.parameters
+        for i in range(len(event.parameters)):
+            event_param_dict[event.parameters[i].name] = i
+        for e in self.sysinfo:
+            if e.name in event_param_dict:
+                event.parameters[event_param_dict[e.name]] = e
+            else:
+                event.parameters.append(e)
```

Comment:
@jinhyunr This feels complex to me. Why not keep it cleaner and simpler: Purge values from `event` that occur in `sysinfo` and then extend the `event` with `sysinfo`. Such logic is harder to get wrong (it entails fewer decisions and, hence, fewer places to be incorrect). Performance is certainly not a concern: We should always code the simplest, correct solution first and then, as driven by data, tweak for performance. (Architectural performance is another matter.)